src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,586 @@
+            }
+
+            function start() {
+                if (!ApplicationInfo.isMobile)
+                    textInput.forceActiveFocus();
+                items.currentLevel = 0
+                initLevel()
+            }
+
+            function initLevel() {
+                score.currentSubLevel = 1
+                score.numberOfSubLevels = dataset[currentLevel].values[1].length
+                initSubLevel()
+            }
+
+            function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    dataset[currentLevel].values[1].splice(randomIndex,1)
+                } else {
+                    questionValue = dataset[currentLevel].values[1][score.currentSubLevel - 1]
+                }
+            }
+
+            function nextLevel() {
+                if(numberOfLevel - 1 == currentLevel ) {
+                    currentLevel = 0
+                } else {
+                    currentLevel++
+                }
+
+                initLevel();
+            }
+
+            function previousLevel() {
+                if(currentLevel == 0) {
+                    currentLevel = numberOfLevel - 1
+                } else {
+                    currentLevel--
+                }
+
+                initLevel();
+            }
+
+            function nextSubLevel() {
+                if(++score.currentSubLevel > score.numberOfSubLevels) {
+                    nextLevel()
+                } else {
+                    initSubLevel();
+                }
+            }
+
+        }
+
+        onStart: {
+            first_screen.visible=true
+            items.start()
+        }
+        onStop: {}
+
+        QtObject {
+            id: morseConverter
+            property string alpha
+            property string morse
+            property string last
+            property var table : {
+             "A" : ".-", "B" : "-...", "C" : "-.-.", "D" : "-..",  "E" : ".", "F" : "..-.", "G" : "--.",
+             "H" : "....", "I" : "..", "J" : ".---", "K" : "-.-",  "L" : ".-..","M" : "--","N" : "-.",
+             "O" : "---", "P" : ".--.", "Q" : "--.-", "R" : ".-.",  "S" : "...","T" : "-", "U" : "..-",
+             "V" : "...-", "W" : ".--",  "X" : "-..-",  "Y" : "-.--","Z" : "--..","1" : ".----","2" : "..---",
+             "3" : "...--",  "4" : "....-", "5" : ".....", "6" : "-....",  "7" : "--...",  "8" : "---..",
+             "9" : "----." , "0" : "-----"
+            };
+            function morse2alpha(str) {
+                var letter=""
+                var input=[]
+                input = str.split(' ')
+                if(input[0]==="")   return ''
+
+                for(var key in table)
+                {
+                    if (table.hasOwnProperty(key)) {
```

Comment:
you are iterating on all the keys of table, why do you need to check if the key is in the table?